src/sys.c

```diff
@@ -731,6 +731,14 @@ DLLEXPORT jl_sym_t* jl_get_OS_NAME()
 #endif
 }
 
+DLLEXPORT jl_sym_t* jl_get_ARCH()
+{
+     static jl_sym_t* ARCH = NULL;
```

Comment:
Thanks, I'll fix the indentation.

It is already tested by `test/compile.jl`, because the load of the `.ji` file will fail if the header doesn't match.  But I should make sure that this actually makes the test fail, rather than just giving a warning.